sdk/core/System.ClientModel/samples/ModelReaderWriter.md

```diff
@@ -53,4 +53,47 @@ string json = @"{
       ""z"": 3
     }";
 OutputModel? model = JsonSerializer.Deserialize<OutputModel>(json, options);
+```
+
+## Using a Proxy
+
+In more advanced scenarios a library user might want to override the behavior of how a model is read or written.
+In this case you can implement your own class which implements the same interface, either `IPersistableModel<T>` or `IJsonModel<T>`, and register it with the `ModelReaderWriterOptions`.
+
+Using a proxy with the following definition
+
+```C# Snippet:Readme_Read_Proxy_ClassStub
+public class OutputModelProxy : IJsonModel<OutputModel>
+```
+
+The example below shows how to read JSON to create a strongly-typed model instance using a proxy.
+
+```C# Snippet:Readme_Read_Proxy
+string json = @"{
+      ""x"": 1,
+      ""y"": 2,
+      ""z"": 3
+    }";
+
+ModelReaderWriterOptions options = new ModelReaderWriterOptions("W");
+options.AddProxy(new OutputModelProxy());
+
+OutputModel? model = ModelReaderWriter.Read<OutputModel>(BinaryData.FromString(json), options);
+```
```

Comment:
It might be helpful to add a final line here (and below) to illustrate the difference that was achieved by using the proxy instead of the model for reading the content.  What happened that was different?  This will help ground the example for readers.